sdk/src/main/scala/com/horizen/SidechainAppModule.scala

```diff
@@ -28,6 +27,9 @@ abstract class SidechainAppModule extends com.google.inject.AbstractModule {
     bind(classOf[TransactionSubmitHelper])
       .to(classOf[TransactionSubmitHelperImpl])
 
+    bind(classOf[AbstractSidechainApp])
+      .to(classOf[SidechainApp])
```

Comment:
why do we need this?
same for account model class

Reply:
This way I was able to make SecretSubmitHelperImpl class work for both Account and UTXO models (`class SecretSubmitHelperImpl @Inject()(val appProvider: Provider[AbstractSidechainApp])`)